src/main/java/core/basesyntax/dao/ReportWorker.java

```diff
@@ -0,0 +1,56 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.fileworker.InputOutputReport;
+import core.basesyntax.service.Validation;
+import java.time.LocalDate;
+import java.util.Map;
+
+public class ReportWorker implements WarehouseDao {
+    private final InputOutputReport inputOutputReport = new InputOutputReport();
```

Comment:
Let's remove this dependecy from this class;
InputOutputReport will get the file name and reads information from file.
This information you can later pass to ReportWorker(it's method)(it's better to use create some models which will represent this information) and base on this information update Your DB(WareHouse);
Then ReportWorker will call the method to get a report from the database; And InputOutputReport will use this report and write it to file;
 For that, you need to create class which will represent a controller and from where you will call your services, pass result of one service and it's method to other service method and so on. It may be class with main method